stage #4/module #5. AWS/theory/lambda.md

```diff
@@ -0,0 +1,124 @@
+- Functions can access AWS services or non-AWS services, AWS services running in VPCs and Non-AWS services running on EC2 instances in an AWS VPC   <br /> [read more](https://docs.aws.amazon.com/lambda/latest/dg/lambda-functions.html)
+- You can request additional memory in 1 MB increments from 128 MB to 10240 MB<br /> [read more](https://docs.aws.amazon.com/lambda/latest/dg/configuration-function-common.html)
+- There is a maximum execution time set - 900 seconds (15 minutes). The default value is 3 seconds<br /> [read more](https://aws.amazon.com/about-aws/whats-new/2018/10/aws-lambda-supports-functions-that-can-run-up-to-15-minutes/)
+- AWS Lambda stores code in Amazon S3 and encrypts it at rest.
+- There are two types of invocation in Lambda: synchronous and asynchronous ones. The key difference between them is that the first one considers obtaining response from the invoked function. The second approach continue processing inbound events without waiting response <br /> [read more](https://docs.aws.amazon.com/lambda/latest/dg/lambda-invocation.html)
+- Lambda is an *event-driven* compute service where AWS Lambda runs code in response to events such as changes to data in an S3 bucket or a DynamoDB table. So, it enables a way not to call a function directly but triggering it with an event (adding an object to SNS for instance) <br /> [read more](https://docs.aws.amazon.com/lambda/latest/dg/invocation-eventsourcemapping.html)
+- There are execution role assigned to a lambda function and resource policy. In order not to get messed within them just remember that execution role is what a function is able to perform while running and resource policy is what service is able to call a function specified <br /> [read more](https://docs.aws.amazon.com/lambda/latest/dg/lambda-intro-execution-role.html)
+- Lambda is able to scale in and scale out automatically in order to handle bursts. Safe limits are 1000 concurrency per region. Maximum up to 3000 depending on a region. It's able to scale by 500 function per minute until the limit is reached or burst is served. <br /> [read more](https://docs.aws.amazon.com/lambda/latest/dg/configuration-concurrency.html)
+- You should differ provisioned and reserved concurrency. The first one guarantees that specified concurrency is served immediately without scaling by 500 functions (mentioned above). The second one is about having a certain number of concurrency prevented from using by another functions within an account <br /> [read more](https://docs.aws.amazon.com/lambda/latest/dg/configuration-concurrency.html)
+- A Lambda function has layers. A layer is a ZIP archive that contains libraries, a custom runtime, or other dependencies.   <br /> [read more](https://docs.aws.amazon.com/lambda/latest/dg/configuration-layers.html) 
```

Comment:
The definition of a layer is actually given 2 times - in key concepts section and right here

Reply:
removed